### PR TITLE
Upsteam DNS enhancement

### PR DESCRIPTION
**Pull request by ChadBHowell**
The purpose of these commits is to do the following:

1) Bring the dnsmasq.conf changes to be inline with how config files are modified on current Debian type systems by using the "*.d" directory. This allows for a cleaner distro upgrade path and avoids stomping our config changes during dnsmasq upgrades.

2) Allow the user to choose between Google and DynDNS upstream servers during the install.  This will later be extended to allow the user to enter custom upstream DNS servers as well.

3) Add unzip to the dependencies to avoid errors on some systems.

### CODE REVIEW

**Comment by dschaper** [comment]
I think there was another issue open, #107 that this addressed. I'll note that this potentially fixes that issue. I agree that we should be using the `*.d` subdirectories. 

I like the choice for user selection up upstreams. And just a quick question, do you have a distro that had unzip/zip missing?

We'll test out the changes and see how things go. Can you set up a PR that merges to the development branch though? We don't like to merge to master, we'd like to keep master as pristine as possible.

Thank you very much for you PR and assistance!

**Comment by jacobsalmela** [comment]
I think this will be an important PR.  I like how you followed the conventions previously introduced by used `sed` to swap in the correct values.  And to second, please make it in a branch other than `master` for stability reasons.

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -230,11 +251,12 @@ $SUDO chmod 755 /usr/local/bin/{gravity,chronometer,whitelist,blacklist,piholeLo
 }
 
 installConfigs(){
-$SUDO mv /etc/dnsmasq.conf /etc/dnsmasq.conf.orig
 $SUDO mv /etc/lighttpd/lighttpd.conf /etc/lighttpd/lighttpd.conf.orig
-$SUDO curl -o /etc/dnsmasq.conf https://raw.githubusercontent.com/jacobsalmela/pi-hole/master/advanced/dnsmasq.conf
+$SUDO curl -o /etc/dnsmasq.conf/01-pihole.conf https://raw.githubusercontent.com/jacobsalmela/pi-hole/master/advanced/01-pihole.conf
```
Comment (Mcat12) [inline automated install/basic-install.sh]: Did you mean to put the config into dnsmasq.d?

Comment (jacobsalmela) [inline automated install/basic-install.sh]: Yeah, I was just wondering that, too.

**Comment by Mcat12** [comment]
It should go in the Devel branch. (Is it just me, or does Devel sound like Devil? :eyes:)

**Comment by ChadBHowell** [comment]
The distro that was missing unzip was the latest Debian image for the Beaglebone Black.  I have fixed the typo and have resubmitted the pull request against the dev branch.

**Comment by Mcat12** [comment]
This pr can be closed now.

**Comment by jacobsalmela** [comment]
Closing per @Mcat12's comment.

**Pull request closed without merging**